Gems/Meshlets/Assets/Shaders/MeshletsCompute.azsl

```diff
@@ -0,0 +1,244 @@
+#pragma once
+
+#include <Atom/Features/SrgSemantics.azsli>
+
+//------------------------------------------------------------------------------
+ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
+{   // [To Do] No needto use StructuredBuffer - change to buffer and test
+    RWStructuredBuffer<int>     m_MeshletsSharedBuffer;
+}
+
+//------------------------------------------------------------------------------
+//! The following meshlet descriptor structure must reflect the structure as 
+//! appeared in MeshletsData.h 
+struct MeshletDescriptor
+{
+    //! Offset into the indirect indices array representing the global index of
+    //! all the meshlet vertices.
+    //! The Indirect vertices array is built as follows:
+    //!     std::vector<uint32_t> indirectIndices;
+    //!     indirectIndices = { { meshlet 1 vertex indices }, { meshlet 2 }, .. { meshlet n} }
+    uint vertexOffset;      // In uint32_t steps
+
+    //! Offset into the global meshlets triangleIndices array represented as:
+    //!     std::vector<uint8_t> triangleIndices;
+    //!     triangleIndices = { {meshlet 1 local indices group}, ... { meshlet n} }
+    //! The local indices are an 8 bits index that can represent up to 256 entries.
+    uint triangleOffset;    // In bytes from the start of the array
+
+    //! Finding a vertex within the meshlet is done like that:
+    //!     triangleOffset = currentMeshlet.triangleOffset + meshletTrIndex * 3;
+    //!     localIndex_i = meshletTriangles[triangleOffset + i];    // i = triangle vertex index 0..2
+    //!     vertexIndex_i =  indirectIndices[currentMeshlet.vertexOffset + localIndex_i];
+
+    //! Amount of vertices and triangle for the mesh - based on this the arrays
+    //! indirectIndices and triangleIndices are created per meshlet.
+    uint vertexCount;
+    uint triangleCount;
```

Comment:
Exactly - not all meshlets can have 64 vertices (or any chosen division), and so this is crucial to maintain the local offset into the array. Without this assumption all hell will break loose since several meshlets will not have the exact amount and they don't come only at the end.